keps/sig-auth/20190916-noderestriction-pods.md

```diff
@@ -174,21 +171,26 @@ makes the following assumptions:
 
 ## Proposal
 
-### Node Restrictions
-
 All restrictions will be enforced through the [NodeRestriction][] admission controller. These
 extensions will be guarded by the `MirrorPodNodeRestriction` feature gate.
 
-On update, only the delta will be restricted. In other words, the Kubelet can modify pods with
-restricted labels / owner references as long as it's not modifying (or adding or
-deleting) the restricted entries.
+Nodes are not granted the `update` or `patch` permissions on pods, but may update `pod/status`. All
+label and owner reference updates will be forbidden through `pod/status` updates, so restrictions
+will not be checked on status updates. In other words, a node _can_ update the status on a pod that
+has un-whitelisted labels.
+
+### Label Restrictions
 
-#### Label Restrictions
+A new reserved annotation will be introduced for namespaces to whitelist mirror-pod labels:
 
-The Kubelet will be prevented from updating pod labels or creating mirror pods with labels, except
-for whitelisted keys:
+```
+node.kubernetes.io/mirror.allowed-label-keys = "key1,key2,..."
+```
 
-- Any labels starting with `unrestricted.node.kubernetes.io/` are allowed.
+When the NodeRestriction controller receives a mirror pod create a request for a node, it will check
+the pod for labels. If it is labeled, the pod's namespace is fetched and checked for the
```

Comment:
Fetched implies live to me. When you write it, can you use a cache instead of live check?

Reply:
Yeah, I assume this is a good case for an informer...